app/controllers/pet_applications_controller.rb

```diff
@@ -0,0 +1,10 @@
+class PetApplicationsController < ApplicationController
+    def create
```

Comment:
Great work using a joins controller here as well to handle creating the joins record